### PR TITLE
Reword docker readme

### PR DESCRIPTION
**Pull request by Emc1992**
**What this PR does / why we need it**:
Makefile doesn't have test-e2e, only test cmd

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

N/A

### CODE REVIEW

File: test/infrastructure/docker/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 CAPD is a reference implementation of an infrastructure provider for the Cluster API project using Docker.
 
+**NOTE:** The Docker provider is **not** designed for production use and is intended for development environments only.
```
Comment (sedefsavas) [inline test/infrastructure/docker/README.md:5]: nit. IMO CAPD goals are actually providing the intend clearly.
```
CAPD Goals
To be a the reference implementation of an infrastructure provider.
The code is highly trusted and used in testing of ClusterAPI.
This provider can be used as a guide for developers looking to implement their own infrastructure provider.
```

File: test/infrastructure/docker/README.md [hunk 2]
```diff
@@ -12,9 +14,11 @@ For a complete overview, please refer to the documentation available [here](http
...
 * This provider can be used as a guide for developers looking to implement their own infrastructure provider.
 
-## End-to-end testing
+## Testing
 
 In order to test your local changes, go to the top level directory of this project, `cluster-api/` and run
-`make -C test/infrastructure/docker test-e2e` to run the test suite.
+`make -C test/infrastructure/docker test` to run the unit tests. 
+
+**Note:** `make test-e2e` runs the CAPI E2E tests that are based on CAPD (CAPD does not have a separated 2e2 suite anymore)
```
Comment (sedefsavas) [inline test/infrastructure/docker/README.md:22]:
```suggestion
**Note:** `make test-e2e` runs the CAPI E2E tests that are based on CAPD (CAPD does not have a separated e2e suite).
```

**Review by sedefsavas — COMMENTED** [review]
Some nits. Otherwise LGTM.

File: test/infrastructure/docker/README.md (hunk 1 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/README.md:5]: Personally, I'm fine if being redundant here, we are adding this sentence consistently across all the docs.

**Comment by fabriziopandini** [comment]
@Emc1992 if you can address @sedefsavas nits, we can get this moving

File: test/infrastructure/docker/README.md (hunk 2 above)
Comment (Emc1992) [reply test/infrastructure/docker/README.md]: corrected

File: test/infrastructure/docker/README.md (hunk 1 above)
Comment (Emc1992) [reply test/infrastructure/docker/README.md:5]: I've left this in - please confirm if this is to be removed / blocking pr merge

**Comment by Emc1992** [comment]
> @Emc1992 if you can address @sedefsavas nits, we can get this moving

I think we are good here?

File: test/infrastructure/docker/README.md (hunk 1 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/README.md:5]: It is ok to leave as it is

**Comment by Emc1992** [comment]
@sedefsavas / @fabriziopandini / @ncdc / @vincepri - Do I need to do anything else here?

**Pull request merged**